extmod/modframebuf.c

```diff
@@ -468,14 +459,167 @@ STATIC mp_obj_t framebuf_line(size_t n_args, const mp_obj_t *args) {
         e += 2 * dy;
     }
 
-    if (0 <= x2 && x2 < self->width && 0 <= y2 && y2 < self->height) {
-        setpixel(self, x2, y2, col);
+    if (0 <= x2 && x2 < fb->width && 0 <= y2 && y2 < fb->height) {
+        setpixel(fb, x2, y2, col);
     }
+}
+
+STATIC mp_obj_t framebuf_line(size_t n_args, const mp_obj_t *args) {
+    (void)n_args;
+
+    mp_obj_framebuf_t *self = MP_OBJ_TO_PTR(args[0]);
+    mp_int_t x1 = mp_obj_get_int(args[1]);
+    mp_int_t y1 = mp_obj_get_int(args[2]);
+    mp_int_t x2 = mp_obj_get_int(args[3]);
+    mp_int_t y2 = mp_obj_get_int(args[4]);
+    mp_int_t col = mp_obj_get_int(args[5]);
+
+    line(self, x1, y1, x2, y2, col);
 
     return mp_const_none;
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(framebuf_line_obj, 6, 6, framebuf_line);
 
+STATIC mp_obj_t framebuf_poly(size_t n_args, const mp_obj_t *args) {
+    (void)n_args;
+
+    mp_obj_framebuf_t *self = MP_OBJ_TO_PTR(args[0]);
+
+    mp_int_t x = mp_obj_get_int(args[1]);
+    mp_int_t y = mp_obj_get_int(args[2]);
+
+    size_t poly_len;
+    mp_obj_t *polygon;
+    mp_obj_get_array(args[3], &poly_len, &polygon);
+
+    if (poly_len == 0) {
+        return mp_const_none;
+    }
+
+    mp_int_t col = mp_obj_get_int(args[4]);
+
+    size_t point_len;
+    mp_obj_t *point;
+
+    mp_obj_get_array(polygon[poly_len - 1], &point_len, &point);
+    if (point_len < 2) {
+        mp_raise_ValueError(MP_ERROR_TEXT("invalid polygon data"));
+    }
+    mp_int_t px1 = mp_obj_get_int(point[0]) + x;
+    mp_int_t py1 = mp_obj_get_int(point[1]) + y;
+
+    for (size_t i = 0; i < poly_len; i++) {
+        mp_obj_get_array(polygon[i], &point_len, &point);
```

Comment:
Better to use `mp_obj_get_array_fixed_n(polygon[i], 2, &point)`, because that will automatically raise an exception if it's not a length-2 list/tuple.

Reply:
Oh I didn't know about mp_obj_get_array_fixed_n -- good tip, thanks

Reply:
I made this change in the latest version of the patchset.